Source/controls/touch/gamepad.cpp

```diff
@@ -0,0 +1,140 @@
+#include <SDL.h>
+
+#include "controls/touch/gamepad.h"
+#include "diablo.h"
+#include "utils/display.h"
+#include "utils/ui_fwd.h"
+
+namespace devilution {
+
+VirtualGamepad VirtualGamepadState;
+
+namespace {
+
+constexpr double Pi = 3.141592653589793;
+
+constexpr bool PointsUp(double angle)
+{
+	constexpr double UpAngle = Pi / 2;
+	constexpr double MinAngle = UpAngle - 3 * Pi / 8;
+	constexpr double MaxAngle = UpAngle + 3 * Pi / 8;
+	return MinAngle <= angle && angle <= MaxAngle;
+}
+
+constexpr bool PointsDown(double angle)
+{
+	constexpr double DownAngle = -Pi / 2;
+	constexpr double MinAngle = DownAngle - 3 * Pi / 8;
+	constexpr double MaxAngle = DownAngle + 3 * Pi / 8;
+	return MinAngle <= angle && angle <= MaxAngle;
+}
+
+constexpr bool PointsLeft(double angle)
+{
+	constexpr double MaxAngle = Pi - 3 * Pi / 8;
+	constexpr double MinAngle = -Pi + 3 * Pi / 8;
+	return !(MinAngle < angle && angle < MaxAngle);
+}
+
+constexpr bool PointsRight(double angle)
+{
+	constexpr double MinAngle = -3 * Pi / 8;
+	constexpr double MaxAngle = 3 * Pi / 8;
+	return MinAngle <= angle && angle <= MaxAngle;
+}
+
+} // namespace
+
+void InitializeVirtualGamepad()
+{
+	int screenPixels = std::min(gnScreenWidth, gnScreenHeight);
+	int inputMargin = screenPixels / 10;
+	int directionPadSize = screenPixels / 4;
+	int padButtonSize = round(1.1 * screenPixels / 10);
+	int padButtonSpacing = inputMargin / 3;
+
+#ifndef USE_SDL1

```

Comment:
Is this file even loaded under SDL1?

Reply:
It was before I overhauled the precompiler checks. Good catch.

Reply:
If you could fix this then I think merging would be a no brainer at this point :)

Reply:
My latest commit fixed it. 👍